### DISCUSSION TITLE
ERROR WHILE PROCESSING RESPONSE(GRBL 1.1F)

### DISCUSSION THREAD

**Discussion by ipsum-coder** [winder/Universal-G-Code-Sender#2397] [Help]
I am currently making up a cnc aurdino plotter working upon grbl 1.1f and it works completely fine until an error pops out of nowhere and the motors stop
```
>>> G2X90.443Y124.956I0.5968J0.4447
ok
>>> G2X91.4665Y126.7274I5.7616J-2.1475
ok
>>> G2X92.0116Y127.0416I0.694J-0.574
ok
>>> G2X92.4957Y126.8999I0.0957J-0.5709
ok
**_Error while processing response <<Run|MPos:93.650,119.610,0.000|F:76Pn:PXYZ>>: For input string: "76Pn:PXYZ"_**
```

I need instant support if someone can help me
the ugs and grbl are up to date versions

**Comment by breiler** [comment]
There are characters missing in the status report:
`<Run|MPos:93.650,119.610,0.000|F:76Pn:PXYZ>`

Should be:
`<Run|MPos:93.650,119.610,0.000|F:76|Pn:PXYZ>`

This is a sign of an unstable USB connection. My guess is that you are running a chinese Ardunio with the serial chip CH340? If that is the case I suggest that you replace it with a higher quality genuine Ardunio.

Or you can try and run run GRBL with a lower bit rate, you will need change [this line in your config](https://github.com/gnea/grbl/blob/master/grbl/config.h#L42) to something like 9600 and recompile the GRBL firmware.

**Reply by ipsum-coder** [reply]
sir the arduino is genuine and i think probably that this might be due to a error in the grbl?

**Reply by breiler** [reply]
Where did you buy the Arduino (links to the store)?

Your problem is not unique but I am quite certain that it isn't related to the GRBL firmware.

Make sure that you are using a high quality USB cable that is shielded and preferably with iron ferrite beads and do not use USB hubs or extensions.

**Reply by ipsum-coder** [reply]
how can i know that my cable is shielded?

**Reply by breiler** [reply]
You have still not provided details about you hardware, so which makes this response longer than it needs to...

If you are using an Arduino Due it is commonly using a USB type B connector. These cables usually have some type of shielding. But if you don't have the specs for it, there is no way of telling. If it looked like this you could safely assume that it is properly shielded:
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/a164f1e7-0961-4725-8605-f2ca9320575b)

If you are using a Arduino Nano it usually uses a USB Mini-B connector. Here you need to be careful as many of these types of cables are used for charging and may not be suited for data transfers.

If it looks like this you should be safe:
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/e78cdbf0-7179-455e-812a-1006391feddb)

If it looks like this, you should try with another:
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/63066bf7-ad9b-4247-afa9-72c854ec4db4)

**Reply by ipsum-coder** [reply]
sir the error has gone when i changed the paud rate in the grbl file to 9600
but 
the plotter just keeps on scratching on the page instead of drawing the figure from the gcode file i am giving to the ugs.
Can you just help if you know anything about it?

**Reply by breiler** [reply]
I can't help without more details about your machine, how you are generating the gcode and what your gcode looks like.
